packages/contracts/test/scripts/MigrateMetapool.t.sol

```diff
@@ -0,0 +1,178 @@
+        staking.removeLiquidity(withdraw, tokens[0]);
+        uint256 v2Bal = v2Metapool.balanceOf(user);
+        v2Metapool.remove_liquidity(v2Bal, [uint256(0), uint256(0)]);
+        vm.stopPrank();
+
+        uint256 userDollarV2 = dollarToken.balanceOf(user) - dollarTokenPreBal;
+        uint256 userLPV2 = curve3Token.balanceOf(user);
+
+        vm.revertTo(snapshot);
+
+        migrate();
+
+        vm.roll(block.number + 10483200);
+        BondingShareV2.Bond memory bondV3 = bonds.getBond(tokens[0]);
+        uint256 withdrawV3 = bondV3.lpAmount;
+        vm.startPrank(user);
+        staking.removeLiquidity(withdrawV3, tokens[0]);
+        uint256 v3Bal = v3Metapool.balanceOf(user);
+
+        v3Metapool.remove_liquidity(v3Bal, [uint256(0), uint256(0)]);
+        vm.stopPrank();
+
+        uint256 userDollarV3 = dollarToken.balanceOf(user) - dollarTokenPreBal;
+        uint256 userLPV3 = curve3Token.balanceOf(user);
+
+        uint256 v3dollarBal = dollarToken.balanceOf(address(v3Metapool));
+        uint256 v3curve3Bal = curve3Token.balanceOf(address(v3Metapool));
+
+        console.log("Metapool LP Tokens withdrawn pre migration: ", v2Bal);
+        console.log("Metapool LP Tokens withdrawn post migration: ", v3Bal);
+        console.log(
+            "UbiquityDollar Tokens withdrawn pre migration: ",
+            userDollarV2
+        );
+        console.log("Curve3 Tokens withdrawn pre migration: ", userLPV2);
+        console.log(
+            "UbiquityDollar Tokens withdrawn post migration: ",
+            userDollarV3
+        );
+        console.log("Curve3 Tokens withdrawn post migration: ", userLPV3);
+        console.log(
+            "Curve Metapool V3 UbiquityDollar balance after withdrawal: ",
+            v3dollarBal
+        );
+        console.log(
+            "Curve Metapool V3 Curve3 balance after withdrawal: ",
+            v3curve3Bal
+        );
+    }
+
+    function migrate()
+        internal
+        returns (uint256 metaBalance, uint256 metaBalanceV3)
+    {
+        vm.startPrank(admin);
+        metaBalance = v2Metapool.balanceOf(address(staking));
+        staking.sendDust(admin, address(v2Metapool), metaBalance);
+
+        v2Metapool.remove_liquidity(metaBalance, [uint256(0), uint256(0)]);
+
+        uint256 curve3Bal = curve3Token.balanceOf(admin);
+        uint256 dollarBal = dollarToken.balanceOf(admin);
+
+        uint256 deposit;
+
+        if (curve3Bal <= dollarBal) {
+            deposit = curve3Bal;
+        } else {
+            deposit = dollarBal;
+        }
+
+        curve3Token.approve(address(v3Metapool), 0);
+        curve3Token.approve(address(v3Metapool), deposit);
+
+        dollarToken.approve(address(v3Metapool), 0);
+        dollarToken.approve(address(v3Metapool), deposit);
+
+        v3Metapool.add_liquidity([deposit, deposit], 0, address(staking));
+        metaBalanceV3 = v3Metapool.balanceOf(address(staking));
+
+        require(metaBalanceV3 > 0);
```

Comment:
We need more than one assertion for the tests.
non exhaustive list of what we want to make sure of : 
 - no tokens are lost during the migrations 
- LP are calculated right 
- LP holders (v2 and v3) are correct